eth/protocols/eth/receipt.go

```diff
@@ -180,6 +221,7 @@ func (r *Receipt) encodeForHash(buf *receiptListBuffers, out *bytes.Buffer) {
 	bloom := r.bloom(&buf.bloom)
 	w.WriteBytes(bloom[:])
 	w.Write(r.Logs)
+	r.maybeWriteDepositFields(w, true)
```

Comment:
LGTM, good alertness with the `encodeForHash` case. Maybe worth a comment for future readers.

Reply:
The test cases that I added luckily failed without this additional logic, so they cover this case.

Added a comment.